addons/hr_holidays/migrations/12.0.1.5/post-migration.py

```diff
@@ -152,21 +152,36 @@ def subscribe_new_subtypes(env):
          'mt_leave_allocation_refused',
          'mt_leave_refused'),
     ]
-    Followers = env['mail.followers']
-    for old_id, new_id1, new_id2 in subtype_mapping:
-        old_mt = env.ref('hr_holidays.' + old_id)
-        followers = Followers.search([('subtype_ids', '=', old_mt.id)])
-        if followers:
-            new_mt1 = env.ref('hr_holidays.' + new_id1)
-            new_mt2 = env.ref('hr_holidays.' + new_id2)
-            subtype_ids = (new_mt1 + new_mt2).ids
-            partner_ids = followers.mapped('partner_id').ids
-            Followers._insert_followers(
-                followers[0].res_model, list(set(followers.mapped('res_id'))),
-                partner_ids,
-                dict((pid, subtype_ids) for pid in partner_ids),
-                [], False, check_existing=True, existing_policy='update',
-            )
+    old_ids, new1_ids, new2_ids = [], [], []
+    for old_id, new1_id, new2_id in subtype_mapping:
+        old_ids.append(env.ref('hr_holidays.' + old_id).id)
+        new1_ids.append(env.ref('hr_holidays.' + new1_id).id)
+        new2_ids.append(env.ref('hr_holidays.' + new2_id).id)
+    for i in range(3):
```

Comment:
Why are you still doing 2 loops with that weird index positioning?

Reply:
Even more... don't do the same SQL (equal except models and ids). Create 2 nested loops that put in the loop variables proper values and just write one SQL.

Reply:
now there is only 1 loop!

Reply:
OK, but see my second comment: you can save one large SQL doing a nested loop including the model in it as variable

Reply:
ok, done. superclean!

Reply:
Great, that's the idea! Isn't better to read this code? :smiley: